packages/convenient_test_dev/lib/src/functions/core.dart

```diff
@@ -52,6 +53,13 @@ class ConvenientTest {
 /// Please make this the only method in your "main" method.
 Future<void> convenientTestMain(ConvenientTestSlot slot, VoidCallback testBody) async {
   myGetIt.registerSingleton<ConvenientTestSlot>(slot);
+
+  await runZonedGuarded(() async {
+    MyIntegrationTestWidgetsFlutterBinding();
```

Comment:
When I create a widget binding here (line 58) it causes the error message:
```Please do *not* initialize `WidgetsBinding.instance` outside `convenientTestMain`.```

You can see the error message in this same file on line 108.

This error happens every time there is a hot reload because convenientTestMain() is called again.
@fzyzcjy

Reply:
Maybe have a look at https://cs.github.com/fzyzcjy/flutter_convenient_test/blob/5837f36965b5dd6599893f0d98ab869baf406d06/packages/convenient_test_dev/lib/src/functions/core.dart?q=Please+do+*not*+initialize+%60WidgetsBinding.instance%60+outside+%60convenientTestMain%60.#L100

Reply:
Yeah I looked through the code there. I tried wrapping my initialization in a `runZoneGuarded` hoping it would tear down the binding after that zone but that did not work.


I read the docs at https://docs.flutter.dev/testing/errors#errors-not-caught-by-flutter but still couldn't figure out how to do it. Honestly I am not sure it is possible to do it in this way.

Its sort of a chicken and an egg problem. 

We need the gRPC server to send over the options to init widgets binding but we can't auto detect the gRPC host address without firstly creating a widgets binding.

If we could somehow tear down the initial widgets binding this could work, but I don't think that is possible.

Can we somehow cache data on the client in between hot reloads? I guess using the file system but that seems over kill and error prone.

Reply:
I see...

May be possible to refactor that widget binding but that will probably be a bit hard.

> Can we somehow cache data on the client in between hot reloads? I guess using the file system but that seems over kill and error prone.

Yes that is too error prone...

Anyway, then maybe just let the users to manually specify IP address...

Reply:
Just added a branch so that this can be configured with --dart-define